source/application/BioXAS/BioXASAppController.cpp

```diff
@@ -67,6 +67,98 @@ void BioXASAppController::shutdown()
 	CLSAppController::shutdown();
 }
 
+void BioXASAppController::onScanEditorCreated(AMGenericScanEditor *editor)
+{
+	onScanEditorCreatedImplementation(editor);
+}
+
+void BioXASAppController::onScanEditorCreatedImplementation(AMGenericScanEditor *editor)
+{
+	if (editor->using2DScanView()) {
+		connect(editor, SIGNAL(dataPositionChanged(AMGenericScanEditor*,QPoint)), this, SLOT(onDataPositionChanged(AMGenericScanEditor*,QPoint)));
+	}
+}
+
+void BioXASAppController::onDataPositionChanged(AMGenericScanEditor *editor, const QPoint &pos)
+{
+	// This should always succeed because the only way to get into this function is using the 2D Generic scan view which currently only is accessed by 2D scans.
+	BioXASGenericStepScanConfiguration *config = qobject_cast<BioXASGenericStepScanConfiguration *>(editor->currentScan()->scanConfiguration());
+
+	if (!config)
+		return;
+
+	QString text;
+	text = QString("Setup at (H, V): (%1 %2, %3 %4)")
+			.arg(editor->dataPosition().x(), 0, 'f', 3)
+			.arg(config->axisControlInfoAt(0).units())
+			.arg(editor->dataPosition().y(), 0, 'f', 3)
+			.arg(config->axisControlInfoAt(1).units());
+
+	QMenu popup(text, editor);
+	QAction *temp = popup.addAction(text);
+	popup.addSeparator();
+	popup.addAction("Go to immediately");
+
+	temp = popup.exec(pos);
+	if (temp){
+		if (temp->text() == "Go to immediately")
+			moveImmediately(editor);
+	}
+}
+
+void BioXASAppController::moveImmediately(const AMGenericScanEditor *editor)
```

Comment:
It is good to break it down for sure and I was planning to do it at the beginning.

As mentioned in the issue, we "think" it might be nicer to have a generalized solution for this so that all the beamlines can benefit from this. I found it is hard to do it since the feature relies on the "type" of the scan configuration. (Ln112). I think it might be better to do the code refactor later.